crates/bevy_render/src/mesh/primitives/dim3/cone.rs

```diff
@@ -0,0 +1,247 @@
+        let num_indices = self.resolution as usize * 6 - 6;
+
+        let mut positions = Vec::with_capacity(num_vertices);
+        let mut normals = Vec::with_capacity(num_vertices);
+        let mut uvs = Vec::with_capacity(num_vertices);
+        let mut indices = Vec::with_capacity(num_indices);
+
+        // Tip
+        positions.push([0.0, half_height, 0.0]);
+
+        // The tip doesn't have a singular normal that works correctly.
+        // We use an invalid normal here so that it becomes NaN in the fragment shader
+        // and doesn't affect the overall shading. This might seem hacky, but it's one of
+        // the only ways to get perfectly smooth cones without creases or other shading artefacts.
+        //
+        // Note that this requires that normals are not normalized in the vertex shader,
+        // as that would make the entire triangle invalid and make the cone appear as black.
+        normals.push([0.0, 0.0, 0.0]);
+
+        // The UVs of the cone are in polar coordinates, so it's like projecting a circle texture from above.
+        // The center of the texture is at the center of the lateral surface, at the tip of the cone.
+        uvs.push([0.5, 0.5]);
+
+        /* Now we build the lateral surface, the side of the cone. */
+
+        // The vertex normals will be perpendicular to the surface.
+        //
+        // Here we get the slope of a normal and use it for computing
+        // the multiplicative inverse of the length of a vector in the direction
+        // of the normal. This allows us to normalize vertex normals efficiently.
+        let normal_slope = self.cone.radius / self.cone.height;
+        // Equivalent to Vec2::new(1.0, slope).length().recip()
+        let normalization_factor = (1.0 + normal_slope * normal_slope).sqrt().recip();
+
+        // How much the angle changes at each step
+        let step_theta = std::f32::consts::TAU / self.resolution as f32;
+
+        // Add vertices for the bottom of the lateral surface.
+        for segment in 0..self.resolution {
+            let theta = segment as f32 * step_theta;
+            let (sin, cos) = theta.sin_cos();
+
+            // The vertex normal perpendicular to the side
+            let normal = Vec3::new(cos, normal_slope, sin) * normalization_factor;
+
+            positions.push([self.cone.radius * cos, -half_height, self.cone.radius * sin]);
+            normals.push(normal.to_array());
+            uvs.push([0.5 + cos * 0.5, 0.5 + sin * 0.5]);
+        }
+
+        // Add indices for the lateral surface. Each triangle is formed by the tip
+        // and two vertices at the base.
+        for j in 1..self.resolution {
+            indices.extend_from_slice(&[0, j + 1, j]);
+        }
+
+        // Close the surface with a triangle between the tip, first base vertex, and last base vertex.
+        indices.extend_from_slice(&[0, 1, self.resolution]);
+
+        /* Now we build the actual base of the cone. */
```

Comment:
nit: [Bevy style guide](https://github.com/bevyengine/bevy/blob/main/.github/contributing/engine_style_guide.md) suggests avoiding block comments.

Reply:
Yeah, I'm aware that normal comments are generally preferred, but these were meant to be kinda like higher level heading comments to make the structure clearer. Using normal comments that don't reference a specific line or block looks a bit weird to me. I'm fine changing it though

Reply:
The existing code base uses line dividers for this
```suggestion
        // ------------------------------------------------------------------------------------------------
        // Build The Base
```